src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NonGCStaticsNode.cs

```diff
@@ -19,10 +19,12 @@ internal class NonGCStaticsNode : ObjectNode, ISymbolNode
     {
         private MetadataType _type;
         private ISymbolNode _classConstructorContext;
+        private bool _hasLazyClassConstructor = false;
```

Comment:
We don't need this field. The original invariant that was used in the HasClassConstructorContext you deleted is still valid...